#166879984 Setup an empty react project

reifred opened the pull request:
### What does this PR do?
Setup an empty react project
### Description of Task to be completed?
Create a blank react project from scratch using Webpack 4, Babel and Airbnb linting option.
### How should this be manually tested?
* After cloning this repo, cd into it and switch to this branch
* run `npm install` command to install project dependencies
* run `npm start` command to launch the application
### Any background context you want to provide?
You must have installed node to run this application.
https://www.webucator.com/how-to/how-install-nodejs-on-mac.cfm
https://treehouse.github.io/installation-guides/windows/node-windows.html
### What are the relevant pivotal tracker stories?
[#166879984](https://www.pivotaltracker.com/story/show/166879984)
### Screenshots.
![image](https://user-images.githubusercontent.com/22131999/60440314-3fce9a80-9c1d-11e9-9d51-6cae2e4af7d7.png)

reifred committed:
chore(setup): setup empty react project

- setup empty react project using webpack and babel
- configure eslint with airbnb

[Delivers #166879984]

Rhytah commented:
> ### How should this be manually tested?
> * After cloning this repo, cd into it and switch to this branch
> * run `npm install` command to install project dependencies
> * run `npm start` command to launch the application
 
@refred you could also include installing `node js` or as a prerequisite just in case does not have it installed already.

reifred commented:
> > ### How should this be manually tested?
> > 
> > * After cloning this repo, cd into it and switch to this branch
> > * run `npm install` command to install project dependencies
> > * run `npm start` command to launch the application
> 
> @refred you could also include installing `node js` or as a prerequisite just in case does not have it installed already.

Thanks @Rhytah, I have updated the PR description and added that in the Background Context section.

Emmanuel-Dominic reviewed (changes requested):
Yeah @reifred, thanks for the great work, however make some changes in the configuration by removing some dependences and tools or lines of settings basically by leaving some that can support the startup of the application and following the story description like:
 - At script then start, the configuration of  the `port 3000` in the package.json file line 8, i think it not necessary since we need to deploy the project on Heroku.
 - Jest which is a tool for testing and so not needed at the startup, so you can do away with it.

Emmanuel-Dominic commented on .eslintrc.json:
```diff
@@ -0,0 +1,41 @@
+    "browser": true,
+    "node": true,
+    "es6": true,
+    "jest": true
```
This testing tool is not required for our startup empty project, so remove it.

reifred replied:
Thanks @Emmanuel-Dominic for your feedback. However, this is not a testing tool in this case. I was specifying environments. If you want to add testing tools you declare them in package.json.
Use this resource on how to configure eslint to gain a more understanding.
https://eslint.org/docs/user-guide/configuring

Emmanuel-Dominic commented on package.json:
```diff
@@ -0,0 +1,45 @@
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "webpack-dev-server --open --hot --port 3000 --mode development",
```
Why this configuration of  the `--port 3000` and `--mode`?  I think its not needed during the Heroku deployment, unless you have some different configuration for us during the deployment. Thank you

reifred replied:
Thanks @Emmanuel-Dominic specifying port and mode is necessary. This is because `npm start` will only be used in the development environment if you want to host that is a different environment so you will make use of `npm run build`.
Kindly use this to gain a more understanding. https://stackoverflow.com/questions/43664200/what-is-the-difference-between-npm-install-and-npm-run-build

Manorlds-Eaglespark commented:
It would be nice to include support for Windows devs as well on how to setup node. Otherwise, everything looks good.
<img width="455" alt="Screenshot 2019-07-02 at 10 55 19" src="https://user-images.githubusercontent.com/28473847/60494730-026b1b00-9cb8-11e9-866d-a73ce771eac3.png">

reifred commented:
> It would be nice to include support for Windows devs as well on how to setup node. Otherwise, everything looks good.
> <img alt="Screenshot 2019-07-02 at 10 55 19" width="455" src="https://user-images.githubusercontent.com/28473847/60494730-026b1b00-9cb8-11e9-866d-a73ce771eac3.png">

Thanks @Manorlds-Eaglespark that has been updated.

Yiga-fred reviewed (approved):
Good work @reifred
